### PR TITLE
chart/sftpgo: Feature/add pv support for sqlite default db

### PR DESCRIPTION
**Pull request by mccrackend**
My attempt at adding a PVC to the chart that addresses adding persistent storage for sqlite db used by default in /var/lib/sftpgo. Am open to feedback and testing from the community!

Addresses #3 #121

### CODE REVIEW

File: charts/sftpgo/templates/pvc.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+kind: PersistentVolumeClaim
+metadata:
+  finalizers:
+  - kubernetes.io/pvc-protection
+  name: {{ include "sftpgo.fullname" . }}-pvc
+  labels:
+    {{- include "sftpgo.labels" . | nindent 4 }}
+spec:
+  {{- toYaml .Values.persistence.pvc | nindent 2 }}
+{{- end }}
```
Comment (sagikazarmark) [inline charts/sftpgo/templates/pvc.yaml:12]: Missing newline at the end of the file

File: charts/sftpgo/templates/pvc.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{{- if .Values.persistence.enabled }}
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  finalizers:
+  - kubernetes.io/pvc-protection
+  name: {{ include "sftpgo.fullname" . }}-pvc
```
Comment (sagikazarmark) [inline charts/sftpgo/templates/pvc.yaml:7]:
```suggestion
  name: {{ include "sftpgo.fullname" . }}
```

File: charts/sftpgo/templates/pvc.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{{- if .Values.persistence.enabled }}
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  finalizers:
+  - kubernetes.io/pvc-protection
```
Comment (sagikazarmark) [inline charts/sftpgo/templates/pvc.yaml:6]:
```suggestion
    - kubernetes.io/pvc-protection
```

File: charts/sftpgo/values.yaml [hunk 4]
```diff
@@ -96,18 +96,19 @@ podAnnotations: {}
 
 # -- Pod [security context](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-pod).
 # See the [API reference](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#security-context) for details.
-podSecurityContext: {}
-  # fsGroup: 2000
+podSecurityContext:
+  fsGroup: 1000
```
Comment (sagikazarmark) [inline charts/sftpgo/values.yaml:100]: Is this a requirement for this change?

File: charts/sftpgo/values.yaml [hunk 5]
```diff
@@ -96,18 +96,19 @@ podAnnotations: {}
...
-securityContext: {}
+securityContext:
   # capabilities:
   #   drop:
   #   - ALL
   # readOnlyRootFilesystem: true
   # runAsNonRoot: true
-  # runAsUser: 1000
+  runAsUser: 1000  # User ID under which your app runs
+  runAsGroup: 1000  # Group ID under which your app runs
```
Comment (sagikazarmark) [inline charts/sftpgo/values.yaml:111]: Is this a requirement for this change?

File: charts/sftpgo/values.yaml [hunk 6]
```diff
@@ -289,3 +290,16 @@ topologySpreadConstraints:
...
   whenUnsatisfiable: DoNotSchedule
+
+# -- Enable persistent storage for the /var/lib/sftpgo directory, saving state of a memory db.
+persistence:
+  enabled: false
+
+  # -- Create the pvc desired specificiation.
+  pvc:
+    accessModes:
+    - ReadWriteOnce
```
Comment (sagikazarmark) [inline charts/sftpgo/values.yaml:301]:
```suggestion
      - ReadWriteOnce
```

File: charts/sftpgo/values.yaml [hunk 7]
```diff
@@ -289,3 +290,16 @@ topologySpreadConstraints:
 
   # -- How to deal with a Pod if it doesn't satisfy the spread constraint.
   whenUnsatisfiable: DoNotSchedule
+
+# -- Enable persistent storage for the /var/lib/sftpgo directory, saving state of a memory db.
```
Comment (sagikazarmark) [inline charts/sftpgo/values.yaml:294]: This should probably be a comment on the enabled key?

File: charts/sftpgo/values.yaml [hunk 8]
```diff
@@ -289,3 +290,16 @@ topologySpreadConstraints:
...
+  enabled: false
+
+  # -- Create the pvc desired specificiation.
+  pvc:
+    accessModes:
+    - ReadWriteOnce
+    resources:
+      requests:
+        storage: 5Gi
+    storageClassName: premium-rwo
```
Comment (sagikazarmark) [inline charts/sftpgo/values.yaml:305]: Is this a sane default (or even required)?

Comment (sagikazarmark) [reply charts/sftpgo/values.yaml:305]: Also: missing newline at the end of the file. Please install editorconfig.

**Review by sagikazarmark — COMMENTED** [review]
Thanks!

Unfortunately, I don't really have the time to test this right now.

Can you please take a look at my comments?

File: charts/sftpgo/values.yaml (hunk 4 above)
Comment (mccrackend) [reply charts/sftpgo/values.yaml:100]: Yes, this was required to give the container access to the attached volume.  Matches the user configuration from the Dockerfile in the sftpgo primary repo.

**Comment by mccrackend** [comment]
Thank you for the review! Comments and changes have been added. 🙏

**Comment by sagikazarmark** [comment]
@mccrackend thanks! Can you please regenerate the readme?

**Review by sagikazarmark — APPROVED** [review]

**Pull request merged**